core/src/packet_deserializer.rs

```diff
@@ -30,6 +28,11 @@ pub struct PacketDeserializer {
     packet_batch_receiver: BankingPacketReceiver,
 }
 
+enum ForEach<'a> {
```

Comment:
My initial thought on this is that it's overly complicated. It seems like the `for_each_packet_batch` fn doesn't really simplify as the match branches could just be replaced by a loop and if statement respectively.

Reply:
yeah, i agree. here's my 2nd attempt to really clean things up: 1ec9956cc3d12b8d2f6cfa8b96eecf023e280297. maybe i should code like this at the time of previous clean up (0952b0db03cd3b046627f268b8c5b74eba5f609d).

Reply:
Love it!